### ISSUE TITLE
nfc-mfclassic block 0 write patch

### ISSUE THREAD

**Issue by GoogleCodeExporter** [timmerk/libnfc#190]
```
Hi,

I've got a special mifare 1k whose block 0 is writeable as any other provided 
you successfully authenticated for the first sector.

Here is a patch for nfc-mfclassic (against current trunk) to try to write block 
0 as a regular block before trying to unlock card with special commands when 
running with W option.

Basically I just added a function to try to write block 0, and a flag 
'write_unlocked' to tell the diff between unlocked mode or not.

Also unlocking attempt had to happen AFTER because the card didn't respond 
after the special unlock command were sent.

Kind regards,

norb



```

Original issue reported on code.google.com by `muda.nor...@gmail.com` on 11 May 2012 at 1:29

Attachments:
- [libnfc-write_block_zero.patch](https://storage.googleapis.com/google-code-attachments/libnfc/issue-190/comment-0/libnfc-write_block_zero.patch)

**Comment by GoogleCodeExporter** [comment]
```
This applies to cards like this:

http://www.xfpga.com/html_products/sp-mf-1k-cpu-26.html

Please include it in current source tree.
```

Original comment by `martin.paljak` on 19 Jan 2013 at 2:26

**Comment by GoogleCodeExporter** [comment]
Original comment by `yob...@gmail.com` on 31 Jan 2013 at 9:46
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision e6ab66037dd7.
```

Original comment by `romu...@libnfc.org` on 5 Apr 2013 at 9:30
- Changed state: **Fixed**
